versions/models.py

```diff
@@ -1071,15 +1076,22 @@ def clone_relations(self, clone, manager_field_name):
             destination.add(item)
 
         # retrieve all current m2m relations pointing the newly created clone
-        m2m_rels = source.through.objects.filter(**{source.source_field.attname: clone.id})  # filter for source_id
+        # filter for source_id
+        m2m_rels = list(source.through.objects.filter(**{source.source_field.attname: clone.id}))
+        later = []
         for rel in m2m_rels:
             # Only clone the relationship, if it is the current one; Simply adjust the older ones to point the old entry
             # Otherwise, the number of pointers pointing an entry will grow exponentially
             if rel.is_current:
-                rel.clone(forced_version_date=self.version_end_date)
+                later.append(rel.clone(forced_version_date=self.version_end_date, in_bulk=True))
             # On rel, set the source ID to self.id
             setattr(rel, source.source_field_name, self)
-            rel.save()
+            if not hasattr(rel, '_not_created') or not rel._not_created:
+                rel.save()
+        # Perform the bulk changes rel.clone() did not perform because of the in_bulk parameter
+        # This saves a huge bunch of SQL queries
+        source.through.objects.filter(id__in=[l.id for l in later]).update(**{'version_start_date': forced_version_date})
+        source.through.objects.bulk_create([r for r in m2m_rels if hasattr(r, '_not_created') and r._not_created])
```

Comment:
I correct myself, I of course was wrong, as it is a ``bulk_create`` and no ``save``, so this could really create duplicate entries. I'm not quite sure that this would be exposed by the current unit tests, we would have to try constructing a test case to provoke it and I'm not quite sure how to do this.